sorting/external_sort.cpp

```diff
@@ -0,0 +1,194 @@
+/**
+ * @file
+ * @brief [External Sorting] https://en.wikipedia.org/wiki/External_sorting
+ * @details
+ * C++ program to sort a file data using external sorting
+ * External sorting is required when the data being sorted do not fit into the
+ * memory (RAM)
+ * @author [Hoai Le](https://github.com/bls-lehoai)
+ */
+
+#include <algorithm>  // for in-memory sorting
+#include <cassert>    // for testing
+#include <fstream>    // for read/write data file
+#include <iostream>   // for I/O operations
+#include <random>     // for generate a random number
+#include <string>     // for string operations (read file line by line)
+
+using namespace std;
+
+constexpr auto BUFF_SIZE = 100000;  ///< size of  buffer data for sorting
+
+/**
+ * Use to hold chunk file information
+ */
+struct ChunkFile {
+    ifstream file;   // file input stream
+    int currentVal;  // current value (in int)
+    string line;     // current line (string)
+};
+
+/**
+ * @brief Prepares 1M rows data (int) data being sorted and write to text file.
+ */
+void mockData(char* path = "data.txt") {
+    cout << "Preparing data" << endl;
+
+    // change to absolutely path if u want
+    ofstream data_file(path, ofstream::trunc);  // overwite
+
+    // use to get a random number
+    mt19937 generator(time(0));
+    uniform_int_distribution<int> distribution(0, 999999);
+
+    for (int i = 0; i < 1000000; i++) {
+        // do random number
+        int randomNumber = distribution(generator);
+        data_file << randomNumber << "\n";
+    }
+    data_file.close();
+
+    cout << "End preparing data" << endl;
```

Comment:
```suggestion
    std::cout << "End preparing data" << std::endl;
```

Reply:
i don't think there are any problems when using namespace std to reduce source code